stdlib/public/core/Substring.swift

```diff
@@ -749,6 +749,39 @@ extension Substring.UTF8View: BidirectionalCollection {
   }
 }
 
+extension Substring.UTF8View {
+
+  @available(SwiftStdlib 6.2, *)
+  public var span: Span<UTF8.CodeUnit> {
+    @lifetime(borrow self)
+    borrowing get {
+#if _runtime(_ObjC)
+      // handle non-UTF8 Objective-C bridging cases here
+      if !_wholeGuts.isFastUTF8 && _wholeGuts._object.hasObjCBridgeableObject {
+        let base: String.UTF8View = self._base
+        let first = base._foreignDistance(from: base.startIndex, to: startIndex)
+        let count = base._foreignDistance(from: startIndex, to: endIndex)
+        let span = unsafe base.span._extracting(
+          unchecked: Range(_uncheckedBounds: (first, first &+ count))
```

Comment:
Again, not objecting, but I do wonder how perf sensitive this path is given the rest of it 😂

Reply:
It makes sense to use the checked `extracting()` here. If we're comfortable with it later, we can switch to unchecked. I was applying the same fast-path logic as in the native case, which is optimistic.

Reply:
The bounds checks would have practically zero cost compared to the two distance calculations that precede them; it makes sense to keep them, as a safeguard against the memoized transcoding somehow getting out of sync with the `NSString` instance (which may be a custom subclass with less-than-perfect mutation semantics).

(I think this reasoning also applies to the bridged cases inside the other members of this type; we should make a FIXME to add proper bounds checking on every bridged path.)

In the native case, the overhead of the checks may be actually measurable, and we generally can trust value semantics to avoid the base getting mutated from under the substring.